// @aws-sdk/client-cloudwatch/dist-es/commands/DeleteMetricStreamCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteMetricStreamInput, DeleteMetricStreamOutput } from "../models/models_0";
import { deserializeAws_queryDeleteMetricStreamCommand, serializeAws_queryDeleteMetricStreamCommand, } from "../protocols/Aws_query";
var DeleteMetricStreamCommand = (function (_super) {
    __extends(DeleteMetricStreamCommand, _super);
    function DeleteMetricStreamCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteMetricStreamCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "DeleteMetricStreamCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteMetricStreamInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteMetricStreamOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteMetricStreamCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteMetricStreamCommand(input, context);
    };
    DeleteMetricStreamCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteMetricStreamCommand(output, context);
    };
    return DeleteMetricStreamCommand;
}($Command));
export { DeleteMetricStreamCommand };
