// @aws-sdk/client-cloudwatch/dist-es/commands/DeleteInsightRulesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteInsightRulesInput, DeleteInsightRulesOutput } from "../models/models_0";
import { deserializeAws_queryDeleteInsightRulesCommand, serializeAws_queryDeleteInsightRulesCommand, } from "../protocols/Aws_query";
var DeleteInsightRulesCommand = (function (_super) {
    __extends(DeleteInsightRulesCommand, _super);
    function DeleteInsightRulesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteInsightRulesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "DeleteInsightRulesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteInsightRulesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteInsightRulesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteInsightRulesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteInsightRulesCommand(input, context);
    };
    DeleteInsightRulesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteInsightRulesCommand(output, context);
    };
    return DeleteInsightRulesCommand;
}($Command));
export { DeleteInsightRulesCommand };
