// @aws-sdk/client-cloudwatch/dist-es/commands/DescribeAlarmHistoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeAlarmHistoryInput, DescribeAlarmHistoryOutput } from "../models/models_0";
import { deserializeAws_queryDescribeAlarmHistoryCommand, serializeAws_queryDescribeAlarmHistoryCommand, } from "../protocols/Aws_query";
var DescribeAlarmHistoryCommand = (function (_super) {
    __extends(DescribeAlarmHistoryCommand, _super);
    function DescribeAlarmHistoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeAlarmHistoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "DescribeAlarmHistoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeAlarmHistoryInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeAlarmHistoryOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeAlarmHistoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeAlarmHistoryCommand(input, context);
    };
    DescribeAlarmHistoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeAlarmHistoryCommand(output, context);
    };
    return DescribeAlarmHistoryCommand;
}($Command));
export { DescribeAlarmHistoryCommand };
