// @aws-sdk/client-cloudwatch/dist-es/commands/DescribeAlarmsForMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeAlarmsForMetricInput, DescribeAlarmsForMetricOutput } from "../models/models_0";
import { deserializeAws_queryDescribeAlarmsForMetricCommand, serializeAws_queryDescribeAlarmsForMetricCommand, } from "../protocols/Aws_query";
var DescribeAlarmsForMetricCommand = (function (_super) {
    __extends(DescribeAlarmsForMetricCommand, _super);
    function DescribeAlarmsForMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeAlarmsForMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "DescribeAlarmsForMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeAlarmsForMetricInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeAlarmsForMetricOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeAlarmsForMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeAlarmsForMetricCommand(input, context);
    };
    DescribeAlarmsForMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeAlarmsForMetricCommand(output, context);
    };
    return DescribeAlarmsForMetricCommand;
}($Command));
export { DescribeAlarmsForMetricCommand };
