// @aws-sdk/client-cloudwatch/dist-es/commands/DisableInsightRulesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisableInsightRulesInput, DisableInsightRulesOutput } from "../models/models_0";
import { deserializeAws_queryDisableInsightRulesCommand, serializeAws_queryDisableInsightRulesCommand, } from "../protocols/Aws_query";
var DisableInsightRulesCommand = (function (_super) {
    __extends(DisableInsightRulesCommand, _super);
    function DisableInsightRulesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisableInsightRulesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "DisableInsightRulesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisableInsightRulesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DisableInsightRulesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisableInsightRulesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDisableInsightRulesCommand(input, context);
    };
    DisableInsightRulesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDisableInsightRulesCommand(output, context);
    };
    return DisableInsightRulesCommand;
}($Command));
export { DisableInsightRulesCommand };
