// @aws-sdk/client-cloudwatch/dist-es/commands/EnableInsightRulesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnableInsightRulesInput, EnableInsightRulesOutput } from "../models/models_0";
import { deserializeAws_queryEnableInsightRulesCommand, serializeAws_queryEnableInsightRulesCommand, } from "../protocols/Aws_query";
var EnableInsightRulesCommand = (function (_super) {
    __extends(EnableInsightRulesCommand, _super);
    function EnableInsightRulesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    EnableInsightRulesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "EnableInsightRulesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableInsightRulesInput.filterSensitiveLog,
            outputFilterSensitiveLog: EnableInsightRulesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableInsightRulesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryEnableInsightRulesCommand(input, context);
    };
    EnableInsightRulesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryEnableInsightRulesCommand(output, context);
    };
    return EnableInsightRulesCommand;
}($Command));
export { EnableInsightRulesCommand };
