// @aws-sdk/client-cloudwatch/dist-es/commands/GetDashboardCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetDashboardInput, GetDashboardOutput } from "../models/models_0";
import { deserializeAws_queryGetDashboardCommand, serializeAws_queryGetDashboardCommand } from "../protocols/Aws_query";
var GetDashboardCommand = (function (_super) {
    __extends(GetDashboardCommand, _super);
    function GetDashboardCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDashboardCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "GetDashboardCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDashboardInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetDashboardOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDashboardCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetDashboardCommand(input, context);
    };
    GetDashboardCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetDashboardCommand(output, context);
    };
    return GetDashboardCommand;
}($Command));
export { GetDashboardCommand };
