// @aws-sdk/client-cloudwatch/dist-es/commands/GetInsightRuleReportCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetInsightRuleReportInput, GetInsightRuleReportOutput } from "../models/models_0";
import { deserializeAws_queryGetInsightRuleReportCommand, serializeAws_queryGetInsightRuleReportCommand, } from "../protocols/Aws_query";
var GetInsightRuleReportCommand = (function (_super) {
    __extends(GetInsightRuleReportCommand, _super);
    function GetInsightRuleReportCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetInsightRuleReportCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "GetInsightRuleReportCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetInsightRuleReportInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetInsightRuleReportOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetInsightRuleReportCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetInsightRuleReportCommand(input, context);
    };
    GetInsightRuleReportCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetInsightRuleReportCommand(output, context);
    };
    return GetInsightRuleReportCommand;
}($Command));
export { GetInsightRuleReportCommand };
