// @aws-sdk/client-cloudwatch/dist-es/commands/GetMetricDataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetMetricDataInput, GetMetricDataOutput } from "../models/models_0";
import { deserializeAws_queryGetMetricDataCommand, serializeAws_queryGetMetricDataCommand, } from "../protocols/Aws_query";
var GetMetricDataCommand = (function (_super) {
    __extends(GetMetricDataCommand, _super);
    function GetMetricDataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetMetricDataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "GetMetricDataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetMetricDataInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetMetricDataOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetMetricDataCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetMetricDataCommand(input, context);
    };
    GetMetricDataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetMetricDataCommand(output, context);
    };
    return GetMetricDataCommand;
}($Command));
export { GetMetricDataCommand };
