// @aws-sdk/client-cloudwatch/dist-es/commands/GetMetricStatisticsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetMetricStatisticsInput, GetMetricStatisticsOutput } from "../models/models_0";
import { deserializeAws_queryGetMetricStatisticsCommand, serializeAws_queryGetMetricStatisticsCommand, } from "../protocols/Aws_query";
var GetMetricStatisticsCommand = (function (_super) {
    __extends(GetMetricStatisticsCommand, _super);
    function GetMetricStatisticsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetMetricStatisticsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "GetMetricStatisticsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetMetricStatisticsInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetMetricStatisticsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetMetricStatisticsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetMetricStatisticsCommand(input, context);
    };
    GetMetricStatisticsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetMetricStatisticsCommand(output, context);
    };
    return GetMetricStatisticsCommand;
}($Command));
export { GetMetricStatisticsCommand };
