// @aws-sdk/client-cloudwatch/dist-es/commands/ListMetricStreamsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListMetricStreamsInput, ListMetricStreamsOutput } from "../models/models_0";
import { deserializeAws_queryListMetricStreamsCommand, serializeAws_queryListMetricStreamsCommand, } from "../protocols/Aws_query";
var ListMetricStreamsCommand = (function (_super) {
    __extends(ListMetricStreamsCommand, _super);
    function ListMetricStreamsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListMetricStreamsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "ListMetricStreamsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListMetricStreamsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListMetricStreamsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListMetricStreamsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListMetricStreamsCommand(input, context);
    };
    ListMetricStreamsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListMetricStreamsCommand(output, context);
    };
    return ListMetricStreamsCommand;
}($Command));
export { ListMetricStreamsCommand };
