// @aws-sdk/client-cloudwatch/dist-es/commands/PutMetricDataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutMetricDataInput } from "../models/models_0";
import { deserializeAws_queryPutMetricDataCommand, serializeAws_queryPutMetricDataCommand, } from "../protocols/Aws_query";
var PutMetricDataCommand = (function (_super) {
    __extends(PutMetricDataCommand, _super);
    function PutMetricDataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutMetricDataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "PutMetricDataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutMetricDataInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutMetricDataCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPutMetricDataCommand(input, context);
    };
    PutMetricDataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPutMetricDataCommand(output, context);
    };
    return PutMetricDataCommand;
}($Command));
export { PutMetricDataCommand };
