// @aws-sdk/client-cloudwatch/dist-es/commands/PutMetricStreamCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutMetricStreamInput, PutMetricStreamOutput } from "../models/models_0";
import { deserializeAws_queryPutMetricStreamCommand, serializeAws_queryPutMetricStreamCommand, } from "../protocols/Aws_query";
var PutMetricStreamCommand = (function (_super) {
    __extends(PutMetricStreamCommand, _super);
    function PutMetricStreamCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutMetricStreamCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "PutMetricStreamCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutMetricStreamInput.filterSensitiveLog,
            outputFilterSensitiveLog: PutMetricStreamOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutMetricStreamCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPutMetricStreamCommand(input, context);
    };
    PutMetricStreamCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPutMetricStreamCommand(output, context);
    };
    return PutMetricStreamCommand;
}($Command));
export { PutMetricStreamCommand };
