// @aws-sdk/client-cloudwatch/dist-es/commands/PutMetricAlarmCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutMetricAlarmInput } from "../models/models_0";
import { deserializeAws_queryPutMetricAlarmCommand, serializeAws_queryPutMetricAlarmCommand, } from "../protocols/Aws_query";
var PutMetricAlarmCommand = (function (_super) {
    __extends(PutMetricAlarmCommand, _super);
    function PutMetricAlarmCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutMetricAlarmCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "PutMetricAlarmCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutMetricAlarmInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutMetricAlarmCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPutMetricAlarmCommand(input, context);
    };
    PutMetricAlarmCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPutMetricAlarmCommand(output, context);
    };
    return PutMetricAlarmCommand;
}($Command));
export { PutMetricAlarmCommand };
