// @aws-sdk/client-cloudwatch/dist-es/commands/SetAlarmStateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetAlarmStateInput } from "../models/models_0";
import { deserializeAws_querySetAlarmStateCommand, serializeAws_querySetAlarmStateCommand, } from "../protocols/Aws_query";
var SetAlarmStateCommand = (function (_super) {
    __extends(SetAlarmStateCommand, _super);
    function SetAlarmStateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetAlarmStateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchClient";
        var commandName = "SetAlarmStateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetAlarmStateInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetAlarmStateCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetAlarmStateCommand(input, context);
    };
    SetAlarmStateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetAlarmStateCommand(output, context);
    };
    return SetAlarmStateCommand;
}($Command));
export { SetAlarmStateCommand };
