// @aws-sdk/client-cloudwatch/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { CloudWatchServiceException as __BaseException } from "./CloudWatchServiceException";
export var AlarmHistoryItem;
(function (AlarmHistoryItem) {
    AlarmHistoryItem.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AlarmHistoryItem || (AlarmHistoryItem = {}));
export var Range;
(function (Range) {
    Range.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Range || (Range = {}));
export var AnomalyDetectorConfiguration;
(function (AnomalyDetectorConfiguration) {
    AnomalyDetectorConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AnomalyDetectorConfiguration || (AnomalyDetectorConfiguration = {}));
export var Dimension;
(function (Dimension) {
    Dimension.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Dimension || (Dimension = {}));
export var Metric;
(function (Metric) {
    Metric.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Metric || (Metric = {}));
export var MetricStat;
(function (MetricStat) {
    MetricStat.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricStat || (MetricStat = {}));
export var MetricDataQuery;
(function (MetricDataQuery) {
    MetricDataQuery.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricDataQuery || (MetricDataQuery = {}));
export var MetricMathAnomalyDetector;
(function (MetricMathAnomalyDetector) {
    MetricMathAnomalyDetector.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricMathAnomalyDetector || (MetricMathAnomalyDetector = {}));
export var SingleMetricAnomalyDetector;
(function (SingleMetricAnomalyDetector) {
    SingleMetricAnomalyDetector.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SingleMetricAnomalyDetector || (SingleMetricAnomalyDetector = {}));
export var AnomalyDetector;
(function (AnomalyDetector) {
    AnomalyDetector.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AnomalyDetector || (AnomalyDetector = {}));
export var AnomalyDetectorType;
(function (AnomalyDetectorType) {
    AnomalyDetectorType["METRIC_MATH"] = "METRIC_MATH";
    AnomalyDetectorType["SINGLE_METRIC"] = "SINGLE_METRIC";
})(AnomalyDetectorType || (AnomalyDetectorType = {}));
export var PartialFailure;
(function (PartialFailure) {
    PartialFailure.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PartialFailure || (PartialFailure = {}));
export var CompositeAlarm;
(function (CompositeAlarm) {
    CompositeAlarm.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CompositeAlarm || (CompositeAlarm = {}));
var ConcurrentModificationException = (function (_super) {
    __extends(ConcurrentModificationException, _super);
    function ConcurrentModificationException(opts) {
        var _this = _super.call(this, __assign({ name: "ConcurrentModificationException", $fault: "client" }, opts)) || this;
        _this.name = "ConcurrentModificationException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ConcurrentModificationException.prototype);
        _this.Message = opts.Message;
        return _this;
    }
    return ConcurrentModificationException;
}(__BaseException));
export { ConcurrentModificationException };
export var DashboardEntry;
(function (DashboardEntry) {
    DashboardEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DashboardEntry || (DashboardEntry = {}));
export var DashboardValidationMessage;
(function (DashboardValidationMessage) {
    DashboardValidationMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DashboardValidationMessage || (DashboardValidationMessage = {}));
var DashboardInvalidInputError = (function (_super) {
    __extends(DashboardInvalidInputError, _super);
    function DashboardInvalidInputError(opts) {
        var _this = _super.call(this, __assign({ name: "DashboardInvalidInputError", $fault: "client" }, opts)) || this;
        _this.name = "DashboardInvalidInputError";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DashboardInvalidInputError.prototype);
        _this.dashboardValidationMessages = opts.dashboardValidationMessages;
        return _this;
    }
    return DashboardInvalidInputError;
}(__BaseException));
export { DashboardInvalidInputError };
var DashboardNotFoundError = (function (_super) {
    __extends(DashboardNotFoundError, _super);
    function DashboardNotFoundError(opts) {
        var _this = _super.call(this, __assign({ name: "DashboardNotFoundError", $fault: "client" }, opts)) || this;
        _this.name = "DashboardNotFoundError";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DashboardNotFoundError.prototype);
        return _this;
    }
    return DashboardNotFoundError;
}(__BaseException));
export { DashboardNotFoundError };
export var Datapoint;
(function (Datapoint) {
    Datapoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Datapoint || (Datapoint = {}));
export var DeleteAlarmsInput;
(function (DeleteAlarmsInput) {
    DeleteAlarmsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAlarmsInput || (DeleteAlarmsInput = {}));
var ResourceNotFound = (function (_super) {
    __extends(ResourceNotFound, _super);
    function ResourceNotFound(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFound", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFound";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFound.prototype);
        return _this;
    }
    return ResourceNotFound;
}(__BaseException));
export { ResourceNotFound };
export var DeleteAnomalyDetectorInput;
(function (DeleteAnomalyDetectorInput) {
    DeleteAnomalyDetectorInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAnomalyDetectorInput || (DeleteAnomalyDetectorInput = {}));
export var DeleteAnomalyDetectorOutput;
(function (DeleteAnomalyDetectorOutput) {
    DeleteAnomalyDetectorOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAnomalyDetectorOutput || (DeleteAnomalyDetectorOutput = {}));
var InternalServiceFault = (function (_super) {
    __extends(InternalServiceFault, _super);
    function InternalServiceFault(opts) {
        var _this = _super.call(this, __assign({ name: "InternalServiceFault", $fault: "server" }, opts)) || this;
        _this.name = "InternalServiceFault";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, InternalServiceFault.prototype);
        _this.Message = opts.Message;
        return _this;
    }
    return InternalServiceFault;
}(__BaseException));
export { InternalServiceFault };
var InvalidParameterCombinationException = (function (_super) {
    __extends(InvalidParameterCombinationException, _super);
    function InvalidParameterCombinationException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidParameterCombinationException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidParameterCombinationException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidParameterCombinationException.prototype);
        return _this;
    }
    return InvalidParameterCombinationException;
}(__BaseException));
export { InvalidParameterCombinationException };
var InvalidParameterValueException = (function (_super) {
    __extends(InvalidParameterValueException, _super);
    function InvalidParameterValueException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidParameterValueException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidParameterValueException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidParameterValueException.prototype);
        return _this;
    }
    return InvalidParameterValueException;
}(__BaseException));
export { InvalidParameterValueException };
var MissingRequiredParameterException = (function (_super) {
    __extends(MissingRequiredParameterException, _super);
    function MissingRequiredParameterException(opts) {
        var _this = _super.call(this, __assign({ name: "MissingRequiredParameterException", $fault: "client" }, opts)) || this;
        _this.name = "MissingRequiredParameterException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, MissingRequiredParameterException.prototype);
        return _this;
    }
    return MissingRequiredParameterException;
}(__BaseException));
export { MissingRequiredParameterException };
var ResourceNotFoundException = (function (_super) {
    __extends(ResourceNotFoundException, _super);
    function ResourceNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFoundException.prototype);
        _this.ResourceType = opts.ResourceType;
        _this.ResourceId = opts.ResourceId;
        _this.Message = opts.Message;
        return _this;
    }
    return ResourceNotFoundException;
}(__BaseException));
export { ResourceNotFoundException };
export var DeleteDashboardsInput;
(function (DeleteDashboardsInput) {
    DeleteDashboardsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDashboardsInput || (DeleteDashboardsInput = {}));
export var DeleteDashboardsOutput;
(function (DeleteDashboardsOutput) {
    DeleteDashboardsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDashboardsOutput || (DeleteDashboardsOutput = {}));
export var DeleteInsightRulesInput;
(function (DeleteInsightRulesInput) {
    DeleteInsightRulesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInsightRulesInput || (DeleteInsightRulesInput = {}));
export var DeleteInsightRulesOutput;
(function (DeleteInsightRulesOutput) {
    DeleteInsightRulesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInsightRulesOutput || (DeleteInsightRulesOutput = {}));
export var DeleteMetricStreamInput;
(function (DeleteMetricStreamInput) {
    DeleteMetricStreamInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteMetricStreamInput || (DeleteMetricStreamInput = {}));
export var DeleteMetricStreamOutput;
(function (DeleteMetricStreamOutput) {
    DeleteMetricStreamOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteMetricStreamOutput || (DeleteMetricStreamOutput = {}));
export var ScanBy;
(function (ScanBy) {
    ScanBy["TIMESTAMP_ASCENDING"] = "TimestampAscending";
    ScanBy["TIMESTAMP_DESCENDING"] = "TimestampDescending";
})(ScanBy || (ScanBy = {}));
export var DescribeAlarmHistoryInput;
(function (DescribeAlarmHistoryInput) {
    DescribeAlarmHistoryInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAlarmHistoryInput || (DescribeAlarmHistoryInput = {}));
export var DescribeAlarmHistoryOutput;
(function (DescribeAlarmHistoryOutput) {
    DescribeAlarmHistoryOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAlarmHistoryOutput || (DescribeAlarmHistoryOutput = {}));
var InvalidNextToken = (function (_super) {
    __extends(InvalidNextToken, _super);
    function InvalidNextToken(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidNextToken", $fault: "client" }, opts)) || this;
        _this.name = "InvalidNextToken";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidNextToken.prototype);
        return _this;
    }
    return InvalidNextToken;
}(__BaseException));
export { InvalidNextToken };
export var DescribeAlarmsInput;
(function (DescribeAlarmsInput) {
    DescribeAlarmsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAlarmsInput || (DescribeAlarmsInput = {}));
export var MetricAlarm;
(function (MetricAlarm) {
    MetricAlarm.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricAlarm || (MetricAlarm = {}));
export var DescribeAlarmsOutput;
(function (DescribeAlarmsOutput) {
    DescribeAlarmsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAlarmsOutput || (DescribeAlarmsOutput = {}));
export var DescribeAlarmsForMetricInput;
(function (DescribeAlarmsForMetricInput) {
    DescribeAlarmsForMetricInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAlarmsForMetricInput || (DescribeAlarmsForMetricInput = {}));
export var DescribeAlarmsForMetricOutput;
(function (DescribeAlarmsForMetricOutput) {
    DescribeAlarmsForMetricOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAlarmsForMetricOutput || (DescribeAlarmsForMetricOutput = {}));
export var DescribeAnomalyDetectorsInput;
(function (DescribeAnomalyDetectorsInput) {
    DescribeAnomalyDetectorsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAnomalyDetectorsInput || (DescribeAnomalyDetectorsInput = {}));
export var DescribeAnomalyDetectorsOutput;
(function (DescribeAnomalyDetectorsOutput) {
    DescribeAnomalyDetectorsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAnomalyDetectorsOutput || (DescribeAnomalyDetectorsOutput = {}));
export var DescribeInsightRulesInput;
(function (DescribeInsightRulesInput) {
    DescribeInsightRulesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInsightRulesInput || (DescribeInsightRulesInput = {}));
export var InsightRule;
(function (InsightRule) {
    InsightRule.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsightRule || (InsightRule = {}));
export var DescribeInsightRulesOutput;
(function (DescribeInsightRulesOutput) {
    DescribeInsightRulesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInsightRulesOutput || (DescribeInsightRulesOutput = {}));
export var DimensionFilter;
(function (DimensionFilter) {
    DimensionFilter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DimensionFilter || (DimensionFilter = {}));
export var DisableAlarmActionsInput;
(function (DisableAlarmActionsInput) {
    DisableAlarmActionsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableAlarmActionsInput || (DisableAlarmActionsInput = {}));
export var DisableInsightRulesInput;
(function (DisableInsightRulesInput) {
    DisableInsightRulesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableInsightRulesInput || (DisableInsightRulesInput = {}));
export var DisableInsightRulesOutput;
(function (DisableInsightRulesOutput) {
    DisableInsightRulesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableInsightRulesOutput || (DisableInsightRulesOutput = {}));
export var EnableAlarmActionsInput;
(function (EnableAlarmActionsInput) {
    EnableAlarmActionsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableAlarmActionsInput || (EnableAlarmActionsInput = {}));
export var EnableInsightRulesInput;
(function (EnableInsightRulesInput) {
    EnableInsightRulesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableInsightRulesInput || (EnableInsightRulesInput = {}));
export var EnableInsightRulesOutput;
(function (EnableInsightRulesOutput) {
    EnableInsightRulesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableInsightRulesOutput || (EnableInsightRulesOutput = {}));
var LimitExceededException = (function (_super) {
    __extends(LimitExceededException, _super);
    function LimitExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "LimitExceededException", $fault: "client" }, opts)) || this;
        _this.name = "LimitExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, LimitExceededException.prototype);
        _this.Message = opts.Message;
        return _this;
    }
    return LimitExceededException;
}(__BaseException));
export { LimitExceededException };
export var GetDashboardInput;
(function (GetDashboardInput) {
    GetDashboardInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDashboardInput || (GetDashboardInput = {}));
export var GetDashboardOutput;
(function (GetDashboardOutput) {
    GetDashboardOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetDashboardOutput || (GetDashboardOutput = {}));
export var GetInsightRuleReportInput;
(function (GetInsightRuleReportInput) {
    GetInsightRuleReportInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetInsightRuleReportInput || (GetInsightRuleReportInput = {}));
export var InsightRuleContributorDatapoint;
(function (InsightRuleContributorDatapoint) {
    InsightRuleContributorDatapoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsightRuleContributorDatapoint || (InsightRuleContributorDatapoint = {}));
export var InsightRuleContributor;
(function (InsightRuleContributor) {
    InsightRuleContributor.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsightRuleContributor || (InsightRuleContributor = {}));
export var InsightRuleMetricDatapoint;
(function (InsightRuleMetricDatapoint) {
    InsightRuleMetricDatapoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsightRuleMetricDatapoint || (InsightRuleMetricDatapoint = {}));
export var GetInsightRuleReportOutput;
(function (GetInsightRuleReportOutput) {
    GetInsightRuleReportOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetInsightRuleReportOutput || (GetInsightRuleReportOutput = {}));
export var LabelOptions;
(function (LabelOptions) {
    LabelOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LabelOptions || (LabelOptions = {}));
export var GetMetricDataInput;
(function (GetMetricDataInput) {
    GetMetricDataInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricDataInput || (GetMetricDataInput = {}));
export var MessageData;
(function (MessageData) {
    MessageData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MessageData || (MessageData = {}));
export var StatusCode;
(function (StatusCode) {
    StatusCode["COMPLETE"] = "Complete";
    StatusCode["INTERNAL_ERROR"] = "InternalError";
    StatusCode["PARTIAL_DATA"] = "PartialData";
})(StatusCode || (StatusCode = {}));
export var MetricDataResult;
(function (MetricDataResult) {
    MetricDataResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricDataResult || (MetricDataResult = {}));
export var GetMetricDataOutput;
(function (GetMetricDataOutput) {
    GetMetricDataOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricDataOutput || (GetMetricDataOutput = {}));
export var GetMetricStatisticsInput;
(function (GetMetricStatisticsInput) {
    GetMetricStatisticsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricStatisticsInput || (GetMetricStatisticsInput = {}));
export var GetMetricStatisticsOutput;
(function (GetMetricStatisticsOutput) {
    GetMetricStatisticsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricStatisticsOutput || (GetMetricStatisticsOutput = {}));
export var GetMetricStreamInput;
(function (GetMetricStreamInput) {
    GetMetricStreamInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricStreamInput || (GetMetricStreamInput = {}));
export var MetricStreamFilter;
(function (MetricStreamFilter) {
    MetricStreamFilter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricStreamFilter || (MetricStreamFilter = {}));
export var MetricStreamOutputFormat;
(function (MetricStreamOutputFormat) {
    MetricStreamOutputFormat["JSON"] = "json";
    MetricStreamOutputFormat["OPEN_TELEMETRY_0_7"] = "opentelemetry0.7";
})(MetricStreamOutputFormat || (MetricStreamOutputFormat = {}));
export var GetMetricStreamOutput;
(function (GetMetricStreamOutput) {
    GetMetricStreamOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricStreamOutput || (GetMetricStreamOutput = {}));
export var GetMetricWidgetImageInput;
(function (GetMetricWidgetImageInput) {
    GetMetricWidgetImageInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricWidgetImageInput || (GetMetricWidgetImageInput = {}));
export var GetMetricWidgetImageOutput;
(function (GetMetricWidgetImageOutput) {
    GetMetricWidgetImageOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetMetricWidgetImageOutput || (GetMetricWidgetImageOutput = {}));
export var ListDashboardsInput;
(function (ListDashboardsInput) {
    ListDashboardsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDashboardsInput || (ListDashboardsInput = {}));
export var ListDashboardsOutput;
(function (ListDashboardsOutput) {
    ListDashboardsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDashboardsOutput || (ListDashboardsOutput = {}));
export var RecentlyActive;
(function (RecentlyActive) {
    RecentlyActive["PT3H"] = "PT3H";
})(RecentlyActive || (RecentlyActive = {}));
export var ListMetricsInput;
(function (ListMetricsInput) {
    ListMetricsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListMetricsInput || (ListMetricsInput = {}));
export var ListMetricsOutput;
(function (ListMetricsOutput) {
    ListMetricsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListMetricsOutput || (ListMetricsOutput = {}));
export var ListMetricStreamsInput;
(function (ListMetricStreamsInput) {
    ListMetricStreamsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListMetricStreamsInput || (ListMetricStreamsInput = {}));
export var MetricStreamEntry;
(function (MetricStreamEntry) {
    MetricStreamEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricStreamEntry || (MetricStreamEntry = {}));
export var ListMetricStreamsOutput;
(function (ListMetricStreamsOutput) {
    ListMetricStreamsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListMetricStreamsOutput || (ListMetricStreamsOutput = {}));
export var ListTagsForResourceInput;
(function (ListTagsForResourceInput) {
    ListTagsForResourceInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceInput || (ListTagsForResourceInput = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var ListTagsForResourceOutput;
(function (ListTagsForResourceOutput) {
    ListTagsForResourceOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceOutput || (ListTagsForResourceOutput = {}));
export var PutAnomalyDetectorInput;
(function (PutAnomalyDetectorInput) {
    PutAnomalyDetectorInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutAnomalyDetectorInput || (PutAnomalyDetectorInput = {}));
export var PutAnomalyDetectorOutput;
(function (PutAnomalyDetectorOutput) {
    PutAnomalyDetectorOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutAnomalyDetectorOutput || (PutAnomalyDetectorOutput = {}));
var LimitExceededFault = (function (_super) {
    __extends(LimitExceededFault, _super);
    function LimitExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "LimitExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "LimitExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, LimitExceededFault.prototype);
        return _this;
    }
    return LimitExceededFault;
}(__BaseException));
export { LimitExceededFault };
export var PutCompositeAlarmInput;
(function (PutCompositeAlarmInput) {
    PutCompositeAlarmInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutCompositeAlarmInput || (PutCompositeAlarmInput = {}));
export var PutDashboardInput;
(function (PutDashboardInput) {
    PutDashboardInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutDashboardInput || (PutDashboardInput = {}));
export var PutDashboardOutput;
(function (PutDashboardOutput) {
    PutDashboardOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutDashboardOutput || (PutDashboardOutput = {}));
export var PutInsightRuleInput;
(function (PutInsightRuleInput) {
    PutInsightRuleInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutInsightRuleInput || (PutInsightRuleInput = {}));
export var PutInsightRuleOutput;
(function (PutInsightRuleOutput) {
    PutInsightRuleOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutInsightRuleOutput || (PutInsightRuleOutput = {}));
export var PutMetricAlarmInput;
(function (PutMetricAlarmInput) {
    PutMetricAlarmInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutMetricAlarmInput || (PutMetricAlarmInput = {}));
export var StatisticSet;
(function (StatisticSet) {
    StatisticSet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StatisticSet || (StatisticSet = {}));
export var MetricDatum;
(function (MetricDatum) {
    MetricDatum.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MetricDatum || (MetricDatum = {}));
export var PutMetricDataInput;
(function (PutMetricDataInput) {
    PutMetricDataInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutMetricDataInput || (PutMetricDataInput = {}));
export var PutMetricStreamInput;
(function (PutMetricStreamInput) {
    PutMetricStreamInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutMetricStreamInput || (PutMetricStreamInput = {}));
export var PutMetricStreamOutput;
(function (PutMetricStreamOutput) {
    PutMetricStreamOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutMetricStreamOutput || (PutMetricStreamOutput = {}));
var InvalidFormatFault = (function (_super) {
    __extends(InvalidFormatFault, _super);
    function InvalidFormatFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidFormatFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidFormatFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidFormatFault.prototype);
        return _this;
    }
    return InvalidFormatFault;
}(__BaseException));
export { InvalidFormatFault };
export var SetAlarmStateInput;
(function (SetAlarmStateInput) {
    SetAlarmStateInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SetAlarmStateInput || (SetAlarmStateInput = {}));
export var StartMetricStreamsInput;
(function (StartMetricStreamsInput) {
    StartMetricStreamsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartMetricStreamsInput || (StartMetricStreamsInput = {}));
export var StartMetricStreamsOutput;
(function (StartMetricStreamsOutput) {
    StartMetricStreamsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartMetricStreamsOutput || (StartMetricStreamsOutput = {}));
export var StopMetricStreamsInput;
(function (StopMetricStreamsInput) {
    StopMetricStreamsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StopMetricStreamsInput || (StopMetricStreamsInput = {}));
export var StopMetricStreamsOutput;
(function (StopMetricStreamsOutput) {
    StopMetricStreamsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StopMetricStreamsOutput || (StopMetricStreamsOutput = {}));
export var TagResourceInput;
(function (TagResourceInput) {
    TagResourceInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceInput || (TagResourceInput = {}));
export var TagResourceOutput;
(function (TagResourceOutput) {
    TagResourceOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceOutput || (TagResourceOutput = {}));
export var UntagResourceInput;
(function (UntagResourceInput) {
    UntagResourceInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceInput || (UntagResourceInput = {}));
export var UntagResourceOutput;
(function (UntagResourceOutput) {
    UntagResourceOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceOutput || (UntagResourceOutput = {}));
